### PR TITLE
fix(share): #MAG-350 fix share rights and view

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
- create folder and create board buttons disappear if no rights in folder
- share right names fixed

## Issue ticket number and link
[MAG-350](https://jira.support-ent.fr/browse/MAG-350)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/common/ShareModal/ShareModal.tsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        <hr />
+        <Heading
+          headingStyle="h4"
+          level="h3"
+          className="mb-16 d-flex align-items-center"
+        >
+          <div className="me-8">{t("magneto.explorer.modal.share.search")}</div>
+          <Tooltip
+            message={
+              "Vos favoris de partage s’affichent en priorité dans votre liste lorsque vous recherchez un groupe ou une personne, vous pouvez les retrouver dans l’annuaire."
```
Comment (niko-amosse) [inline frontend/src/common/ShareModal/ShareModal.tsx:241]: i18n ?

**Review by niko-amosse — APPROVED** [review]

File: frontend/src/utils/share.utils.tsx
Comment (FlorentMr) [inline frontend/src/utils/share.utils.tsx:1]: On a déjà ce fichier dans frontend/src/services/utils, faut voir si on supprime l'ancien ou si on supprime le nouveau

**Review by FlorentMr — APPROVED** [review]

**Review by FlorentMr — CHANGES_REQUESTED** [review]
Check quels fichiers garder

File: frontend/.eslintrc.cjs [hunk 2]
```diff
@@ -23,6 +23,7 @@ module.exports = {
     "@typescript-eslint/no-non-null-assertion": "off",
     "@typescript-eslint/ban-ts-comment": "off",
     "@typescript-eslint/ban-types": "off",
+    "react-hooks/exhaustive-deps": "off",
```
Comment (alicedraillard) [inline frontend/.eslintrc.cjs:26]: handle missing dependencies warning

File: frontend/src/utils/share.utils.ts
Comment (alicedraillard) [inline frontend/src/utils/share.utils.ts:1]: file moved here

**Review by FlorentMr — APPROVED** [review]

**Pull request merged**